### PR TITLE
Global documentation and CSS update

### PR DESCRIPTION
**Pull request by RedGui**
This pull features another set of major and minor changes alongside a major stylesheet cleanup and revamp (See video below)

[<b>&#128267;</b> Stylesheet reference videos](https://drive.google.com/drive/folders/0ByiupY2x-_RfdTlYMmZ5WXc4d1k?usp=sharing)

This merge entails a documentation CSS update. The newer documentation stylesheet is based on normalize 3.0, whereas the current documentation in PR is running on a more messy normalize 2.0, which is causing some elements to appear not as intended (i.e buttons in the section models). Please update the stylesheet directory with this merge

**Important:** Some of the slate dependencies used for creating the PR version of the docu are outdated or not set up to allow relative links with "-". This is causing a multitude of more complex section reference links not to function at all (i.e links from the Camera Model section). Please update the dependencies or double-check the settings before building the new PR documentation

In addition to the regular stylesheet another is shown in one of the videos, showing a completely different approach - with bright, positive colors. Currently just a demo, however if it gathers enough interest, this work in progress could be extended and implemented

Global cross-section edit including:

- General review (minor content and major structure improvements for all sections),
- Content update (for AAA, Account, Action, Annotation, Asset, Authentication, Bridge, Images and Video, Camera, Feedback, Layout, Managed Switch, Metric, Terms of Service, PNG Span, Poll, Types, User, Introduction, How To’s and Example Code)
- Account content update based on the new camera_shares structure with camera_share_perms
- Search and Recording sections have been hidden (due to missing information)
- Major content update to Annotations including cURL requests, HTTP requests/responses, parameter tables etc.
- Major content update to Managed Switch including cURL requests, HTTP requests/responses, parameter tables etc.

### CODE REVIEW

**Comment by georgekarrv** [comment]
root@ee7ddeebd523:/usr/src/app# gem list

*** LOCAL GEMS ***

activesupport (4.1.16)
bigdecimal (default: 1.3.0)
bundler (1.15.3)
celluloid (0.16.0)
chunky_png (1.3.8)
coffee-script (2.4.1)
coffee-script-source (1.12.2)
compass (1.0.3)
compass-core (1.0.3)
compass-import-once (1.0.5)
did_you_mean (1.1.0)
em-websocket (0.5.1)
erubis (2.7.0)
eventmachine (1.2.5)
execjs (2.7.0)
ffi (1.9.18)
haml (5.0.1)
hike (1.2.3)
hitimes (1.2.5)
hooks (0.4.1)
http_parser.rb (0.6.0)
i18n (0.7.0)
io-console (default: 0.4.6)
json (default: 2.0.2, 1.8.6)
kramdown (1.14.0)
libv8 (3.16.14.19 x86_64-linux)
listen (2.10.1)
middleman (3.3.12)
middleman-core (3.3.12)
middleman-gh-pages (0.3.1)
middleman-livereload (3.3.4)
middleman-sprockets (3.5.0)
middleman-syntax (2.1.0)
minitest (5.10.3, 5.10.1)
multi_json (1.12.1)
net-telnet (0.1.1)
openssl (default: 2.0.3)
padrino-helpers (0.12.8.1)
padrino-support (0.12.8.1)
power_assert (0.4.1)
psych (default: 2.2.2)
rack (1.6.8)
rack-livereload (0.3.16)
rack-test (0.6.3)
rake (12.0.0, 10.4.2)
rb-fsevent (0.10.2)
rb-inotify (0.9.10)
rdoc (default: 5.0.0)
redcarpet (3.2.3)
ref (2.0.0)
rouge (1.7.2)
rubygems-update (2.6.12)
sass (3.4.25)
sprockets (2.12.4)
sprockets-helpers (1.1.0)
sprockets-sass (1.3.1)
temple (0.8.0)
test-unit (3.2.3)
therubyracer (0.12.3)
thor (0.19.4)
thread_safe (0.3.6)
tilt (1.4.1)
timers (4.0.4)
tzinfo (1.2.3)
uber (0.0.15)
uglifier (2.7.2)
xmlrpc (0.2.1)

What is behind? isn't 'slate' = 'middleman' which seems to be 3.3.0 already...

**Comment by georgekarrv** [comment]
modiy in Gemfile to get correct dependency versions

**Comment by JeffryRatcliffEagleEye** [comment]
This file is setting the dependence for the docker image
https://github.com/EENCloud/api-docs/blob/master/Gemfile

**Comment by RedGui** [comment]
The docker image was behaving a little unpredictable for us. Additionally, after updating middleman it sets the server on localhost, which is unreachable from the outside, but with the original gemfile version it doesn't even complete the command.

Here's the list of gems from our environment (where the hyphenated links are working):

*** LOCAL GEMS ***

activesupport (5.0.4, 5.0.1)
addressable (2.5.1, 2.5.0)
autoprefixer-rails (6.7.7.2, 6.6.1)
backports (3.8.0, 3.6.8)
bigdecimal (1.2.8)
bundler (1.14.6)
coffee-script (2.4.1)
coffee-script-source (1.12.2)
compass-import-once (1.0.5)
concurrent-ruby (1.0.5, 1.0.4)
contracts (0.13.0)
did_you_mean (1.0.0)
dotenv (2.2.1, 2.2.0)
erubis (2.7.0)
execjs (2.7.0)
fast_blank (1.0.0)
fastimage (2.1.0, 2.0.1)
ffi (1.9.18, 1.9.17)
haml (5.0.1, 4.0.7)
hamster (3.0.0)
hashie (3.5.6, 3.5.1)
i18n (0.7.0)
io-console (0.4.5)
json (1.8.3)
kramdown (1.14.0, 1.13.2)
listen (3.0.8)
memoist (0.16.0, 0.15.0)
middleman (4.2.1)
middleman-autoprefixer (2.7.1)
middleman-cli (4.2.1)
middleman-core (4.2.1)
middleman-sprockets (4.1.0)
middleman-syntax (3.0.0)
minitest (5.10.3, 5.10.1, 5.8.4)
net-telnet (0.1.1)
padrino-helpers (0.13.3.4, 0.13.3.3)
padrino-support (0.13.3.4, 0.13.3.3)
parallel (1.12.0, 1.10.0)
power_assert (0.2.7)
psych (2.0.17)
public_suffix (2.0.5)
rack (2.0.3, 2.0.1)
rake (10.5.0)
rb-fsevent (0.10.2, 0.9.8)
rb-inotify (0.9.10, 0.9.8)
rdoc (4.2.1)
redcarpet (3.4.0)
rouge (2.0.7)
sass (3.4.25, 3.4.23)
servolux (0.13.0, 0.12.0)
sprockets (3.7.1)
temple (0.8.0)
test-unit (3.1.7)
thor (0.19.4)
thread_safe (0.3.6, 0.3.5)
tilt (2.0.8, 2.0.6)
tzinfo (1.2.3, 1.2.2)
uglifier (3.2.0, 3.0.4)

Sent an email to @JeffryRatcliffEagleEye with the details.

**Comment by RedGui** [comment]
Updated the docker and gem files. Docker-compose up will first build the files and then start middleman on port 4567 (which is being forwarded).

This should resolve the issues. Tested on 2 devices so far. No errors or warnings during execution.

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM ruby:2
+FROM ruby:2.3.1
 
 WORKDIR /usr/src/app
 COPY Gemfile ./
 RUN bundle install
 
 ADD . .
 EXPOSE 4567
-CMD ["bundle", "exec", "middleman", "server"]
```
Comment (JeffryRatcliffEagleEye) [inline Dockerfile]: this needs to be in your docker-compose.override.yml

File: docker-compose.yml [hunk 2]
```diff
@@ -5,4 +5,11 @@
      volumes:
        - .:/usr/src/app
      ports:
-       - "8000:4567"
```
Comment (JeffryRatcliffEagleEye) [inline docker-compose.yml]: this need stay the same, production expects this port to be working

File: docker-compose.yml [hunk 3]
```diff
@@ -5,4 +5,11 @@
...
+       - "4567:4567"
```
Comment (JeffryRatcliffEagleEye) [inline docker-compose.yml]: these changes need to be you docker-compose.override.yml

**Comment by RedGui** [comment]
Added Middleman-Livereload and implemented the review changes

File: docker-compose.yml [hunk 4]
```diff
@@ -6,3 +6,14 @@
        - .:/usr/src/app
      ports:
        - "8000:4567"
+       - "4567-4568:4567-4568"
```
Comment (JeffryRatcliffEagleEye) [inline docker-compose.yml]: This need to be in override, we do not want to expose these ports

**Comment by JeffryRatcliffEagleEye** [comment]
Move exposing configuration and middle man to your override yml, this file should not be checked in
docker-compose.override.yml

**Comment by RedGui** [comment]
Moved the volume, port and command configuration out of docker-compose.yml and into docker-compose.override.yml

**Comment by JeffryRatcliffEagleEye** [comment]
I have delete docker-compose.override.yml, this file should not be checked in, otherwise it will used in production.

I also added a readme for docker

Here are the contents of the override file
docker-compose.override.yml 
 version: '2'
 services:
   apidoc:
     volumes:
       - .:/usr/src/app

     ports:
       - "4567-4568:4567-4568"

     command: /bin/bash -c "echo ==================== B U I L D ====================;
                            pwd;
                            bundle exec middleman build --clean;
                            chmod 777 -R build build/* Gemfile.lock;
                            echo;
                            echo =================== S E R V E R ===================;
                            bundle exec middleman server;
                            echo;
                            echo ==================== A B O R T ===================="

**Comment by JeffryRatcliffEagleEye** [comment]
:+1:

**Pull request merged**